### DISCUSSION TITLE
Best approach of Oqtane Module Development

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#1141] [General]
Hi,

What is the best approach of Oqtane Module Development?

I have following modules which is required to reference DLLs each other, 

1. Inventory -> Catalog
2. Purchase -> Purchase Order & Supplier
3. Sales -> Sales Order & Customer
4. Generic Extensions & Helpers

Also, I'm considering multiple database support such as MSSQL, MySQL and SQLite.

Thanks & Regards!

**Comment by leigh-pointer** [comment]
You first create your modules separately then add them all to one solution. 
In the module projects you can reference, by project the modules you require in the solution.
This will ensure you have no circular references.

**Comment by hishamco** [comment]
There are two approaches for module development 1. Internal which adds your module in the Oqtane solution 2. External which adds your module into a separate solution, both ways should work the same

> Also, I'm considering multiple database support such as MSSQL, MySQL and SQLite.

Multiple database support is one of the requested features for `2.1.0` please refer to https://github.com/oqtane/oqtane.framework/discussions/964
